api/envoy/config/trace/v2/trace.proto

```diff
@@ -61,12 +61,30 @@ message Tracing {
 // Configuration for the LightStep tracer.
 // [#extension: envoy.tracers.lightstep]
 message LightstepConfig {
+  // Available propagation modes
+  enum PropagationMode {
+    // Propagate trace context in the single header x-ot-span-context.
+    ENVOY = 0;
+
+    // Propagate trace context using LightStep's native format.
+    LIGHTSTEP = 1;
+
+    // Propagate trace context using the b3 format.
+    B3 = 2;
+
+    // Propagation trace context using the w3 trace-context standard.
+    TRACE_CONTEXT = 3;
```

Comment:
Is there a chance that now or in the future any of these formats requires additional parameters? If so, I would favor a `oneof` with a message type per format to future proof.

Reply:
They shouldn't require more arguments. They correspond to a similar [enum](https://github.com/lightstep/lightstep-tracer-cpp/blob/master/include/lightstep/tracer.h#L25) in the tracer's options.